src/flirtlib.cpp

```diff
@@ -94,6 +129,41 @@ FlirtlibFeatures::extractFeatures (sm::LaserScan::ConstPtr scan) const
   return pts;
 }
 
+InterestPointVec
+FlirtlibFeatures::extractFeatures (const sensor_msgs::LaserScan& scan, const tf::StampedTransform& transform) const
+{
+  boost::shared_ptr<LaserReading> reading = fromRos(scan);
+  InterestPointVec pts;
+  detector_->detect(*reading, pts);
+  BOOST_FOREACH (InterestPoint* p, pts)
+  {
+    p->setDescriptor(descriptor_->describe(*p, *reading));
```

Comment:
Up to this point the code if this method is the same as the previous one.

It's not important, but if you want to avoid duplicating code, you can create a separate method/function that takes a `InterestPointVec& pts` and transform them.